lib/app/migrate.ts

```diff
@@ -0,0 +1,584 @@
+      derivedAccountId,
+      projectName,
+      buildId
+    );
+
+    const absoluteDestPath = projectDest
+      ? path.resolve(getCwd(), projectDest)
+      : getCwd();
+
+    await extractZipArchive(
+      zippedProject,
+      sanitizeFileName(projectName),
+      path.resolve(absoluteDestPath),
+      { includesRootDir: false, hideLogs: false }
+    );
+
+    SpinniesManager.succeed('fetchingMigratedProject', {
+      text: i18n(
+        `commands.project.subcommands.migrateApp.spinners.downloadingProjectContentsComplete`
+      ),
+    });
+
+    logger.success(`Saved ${projectName} to ${projectDest}`);
+  } catch (error) {
+    SpinniesManager.fail('fetchingMigratedProject', {
+      text: i18n(
+        `commands.project.subcommands.migrateApp.spinners.downloadingProjectContentsFailed`
+      ),
+    });
+    throw error;
+  }
+}
+
+export async function migrateApp2025_2(
+  derivedAccountId: number,
+  options: ArgumentsCamelCase<MigrateAppOptions>
+) {
+  SpinniesManager.init();
+
+  const { appIdToMigrate, projectName, projectDest } =
+    await handleMigrationSetup(derivedAccountId, options);
+
+  if (!appIdToMigrate) {
+    return;
+  }
+
+  const { migrationId, uidMap } = await handleMigrationProcess(
+    derivedAccountId,
+    appIdToMigrate
+  );
+
+  const buildId = await finalizeMigration(
+    derivedAccountId,
+    migrationId,
+    uidMap,
+    projectName,
+    options.platformVersion
+  );
+
+  if (!buildId) {
+    throw new Error('Migration Failed');
+  }
+
+  await downloadProjectFiles(
+    derivedAccountId,
+    projectName,
+    buildId,
+    projectDest
+  );
+}
+
+export function logInvalidAccountError(i18nKey: string) {
+  uiLine();
+  logger.error(i18n(`${i18nKey}.errors.invalidAccountTypeTitle`));
+  logger.log(
+    i18n(`${i18nKey}.errors.invalidAccountTypeDescription`, {
+      useCommand: uiCommandReference('hs accounts use'),
+      authCommand: uiCommandReference('hs auth'),
+    })
+  );
+  uiLine();
+}
+
+export async function migrateApp2023_2(
+  derivedAccountId: number,
+  options: ArgumentsCamelCase<MigrateAppOptions>,
+  accountConfig: CLIAccount
+) {
+  const i18nKey = 'commands.project.subcommands.migrateApp';
+  const accountName = uiAccountDescription(derivedAccountId);
+
+  if (!isAppDeveloperAccount(accountConfig)) {
+    logInvalidAccountError(i18nKey);
+    process.exit(EXIT_CODES.SUCCESS);
+  }
+
+  const { appId } =
+    'appId' in options
+      ? options
+      : await selectPublicAppPrompt({
+          accountId: derivedAccountId,
+          accountName,
+          isMigratingApp: true,
+        });
+
+  try {
+    const { data: selectedApp } = await fetchPublicAppMetadata(
+      appId,
+      derivedAccountId
+    );
+    // preventProjectMigrations returns true if we have not added app to allowlist config.
+    // listingInfo will only exist for marketplace apps
+    const preventProjectMigrations = selectedApp.preventProjectMigrations;
```

Comment:
Are we reusing the `preventProjectMigrations` property from the last migration cycle?

Reply:
No, we aren't reusing much which is why it was split into multiple methods.